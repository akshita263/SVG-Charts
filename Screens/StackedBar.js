import React from 'react'
import { StackedBarChart } from 'react-native-svg-charts'
import Header from '../Component/Header';
import { StyleSheet, View, Text, TouchableOpacity, ScrollView } from 'react-native'
 

export default class StackedBarEx extends React.PureComponent {
    render() {
        const data = [
            {
                month: new Date(2015, 0, 1),
                apples: 3840,
                bananas: 1920,
                cherries: 960,
                dates: 400,
            },
            {
                month: new Date(2015, 1, 1),
                apples: 1600,
                bananas: 1440,
                cherries: 960,
                dates: 400,
            },
            {
                month: new Date(2015, 2, 1),
                apples: 640,
                bananas: 960,
                cherries: 3640,
                dates: 400,
            },
            {
                month: new Date(2015, 3, 1),
                apples: 3320,
                bananas: 480,
                cherries: 640,
                dates: 400,
            },
        ]
        const colors = ['#03396c', '#005b96', '#6497b1', '#b3cde0']
        const keys = ['apples', 'bananas', 'cherries', 'dates']

        return (
            <ScrollView>
                <Text style={styles.title}>
                    STACKED BAR CHART: VERTICAL
                </Text>
                <TouchableOpacity style={styles.chart}>
                    <StackedBarChart
                        style={{ height: 300 }}
                        keys={keys}
                        colors={colors}
                        data={data}
                        showGrid={false}
                        contentInset={{ top: 10, bottom: 10 }}
                    />
                </TouchableOpacity>
                <Text style={styles.title}>
                    STACKED BAR CHART: HORIZONTAL
                </Text>
                <TouchableOpacity style={styles.chart}>
                    <StackedBarChart
                        style={{ height: 300 }}
                        keys={keys}
                        colors={colors}
                        data={data}
                        showGrid={false}
                        horizontal={true}
                        contentInset={{ left: 10, right: 10 }}
                    />
                </TouchableOpacity>
            </ScrollView>
        )
    }
}

const styles = StyleSheet.create({
    chart:{
        margin: 20,
        borderWidth: 5,
        padding: 8,
    },
    title: {
        fontSize: 30,
        padding:20,
        alignSelf: 'center',
    }
});

