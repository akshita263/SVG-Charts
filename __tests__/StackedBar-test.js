import React from 'react';
import 'react-native';
import StackedBar from '../Screens/StackedBar';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<StackedBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});