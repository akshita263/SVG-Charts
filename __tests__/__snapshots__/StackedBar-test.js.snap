// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 30,
          "padding": 20,
        }
      }
    >
      STACKED BAR CHART: VERTICAL
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderWidth": 5,
          "margin": 20,
          "opacity": 1,
          "padding": 8,
        }
      }
    >
      <View
        style={
          Object {
            "height": 300,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 30,
          "padding": 20,
        }
      }
    >
      STACKED BAR CHART: HORIZONTAL
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderWidth": 5,
          "margin": 20,
          "opacity": 1,
          "padding": 8,
        }
      }
    >
      <View
        style={
          Object {
            "height": 300,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
